extensions/standard-processors/tests/unit/EvaluateJsonPathTests.cpp

```diff
@@ -0,0 +1,330 @@
+
+  CHECK(controller_.plan->getContent(result_flow_file) == "{}");
+
+  for (const auto& [key, value] : expected_attributes) {
+    std::string attribute_value;
+    if (!expect_attributes) {
+      CHECK_FALSE(result_flow_file->getAttribute(key, attribute_value));
+    } else {
+      CHECK(result_flow_file->getAttribute(key, attribute_value));
+      CHECK(attribute_value == value);
+    }
+  }
+
+  if (warn_path_not_found_behavior) {
+    CHECK(utils::verifyLogLinePresenceInPollTime(0s, "JSON path '$.firstName' not found for attribute key 'attribute1'"));
+    CHECK(utils::verifyLogLinePresenceInPollTime(0s, "JSON path '$.lastName' not found for attribute key 'attribute2'"));
+  }
+}
+
+TEST_CASE_METHOD(EvaluateJsonPathTestFixture, "JSON paths are not found in content when destination is set in content", "[EvaluateJsonPathTests]") {
+  controller_.plan->setProperty(evaluate_json_path_processor_, processors::EvaluateJsonPath::Destination, "flowfile-content");
+  controller_.plan->setDynamicProperty(evaluate_json_path_processor_, "attribute", "$.firstName");
+
+  bool warn_path_not_found_behavior = false;
+  SECTION("Ignore path not found behavior") {
+    controller_.plan->setProperty(evaluate_json_path_processor_, processors::EvaluateJsonPath::PathNotFoundBehavior, "ignore");
+  }
+
+  SECTION("Skip path not found behavior") {
+    controller_.plan->setProperty(evaluate_json_path_processor_, processors::EvaluateJsonPath::PathNotFoundBehavior, "skip");
+  }
+
+  SECTION("Warn path not found behavior") {
+    controller_.plan->setProperty(evaluate_json_path_processor_, processors::EvaluateJsonPath::PathNotFoundBehavior, "warn");
+    warn_path_not_found_behavior = true;
+  }
+
+  auto result = controller_.trigger({{.content = "{}"}});
+
+  REQUIRE(result.at(processors::EvaluateJsonPath::Matched).empty());
+  REQUIRE(result.at(processors::EvaluateJsonPath::Unmatched).size() == 1);
+  REQUIRE(result.at(processors::EvaluateJsonPath::Failure).empty());
+
+  const auto result_flow_file = result.at(processors::EvaluateJsonPath::Unmatched).at(0);
+
+  CHECK(controller_.plan->getContent(result_flow_file) == "{}");
+
+  std::string attribute_value;
+  CHECK_FALSE(result_flow_file->getAttribute("attribute", attribute_value));
+
+  if (warn_path_not_found_behavior) {
+    CHECK(utils::verifyLogLinePresenceInPollTime(0s, "JSON path '$.firstName' not found for attribute key 'attribute'"));
+  }
+}
+
+TEST_CASE_METHOD(EvaluateJsonPathTestFixture, "JSON path query result does not match the required return type", "[EvaluateJsonPathTests]") {
+  controller_.plan->setDynamicProperty(evaluate_json_path_processor_, "attribute", "$.name");
+
+  SECTION("Return type is set to scalar automatically when destination is set to flowfile-attribute") {
+    controller_.plan->setProperty(evaluate_json_path_processor_, processors::EvaluateJsonPath::Destination, "flowfile-attribute");
+  }
```

Comment:
Is this SECTION missing a pair, where the `Destination` is `flowfile-content` and the `Return Type` is `scalar`?  A SECTION by itself is just a block.